src/hotspot/share/gc/shenandoah/shenandoahEvacTracker.cpp

```diff
@@ -89,24 +99,27 @@ void ShenandoahEvacuationTracker::print_evacuations_on(outputStream* st,
   mutators->print_on(st);
   st->cr();
 
-  AgeTable young_region_ages(false), old_region_ages(false);
   ShenandoahHeap* heap = ShenandoahHeap::heap();
-  for (uint i = 0; i < heap->num_regions(); ++i) {
-    ShenandoahHeapRegion* r = heap->get_region(i);
-    if (r->age() > 0 && r->age() < AgeTable::table_size) {
-      if (r->is_young()) {
-        young_region_ages.add(r->age(), r->get_live_data_words());
-      } else {
-        assert(r->is_old(), "Sanity");
-        old_region_ages.add(r->age(), r->get_live_data_words());
+  if (_generational) {
+    AgeTable young_region_ages(false), old_region_ages(false);
+    for (uint i = 0; i < heap->num_regions(); ++i) {
+      ShenandoahHeapRegion* r = heap->get_region(i);
+      if (r->age() > 0 && r->age() < AgeTable::table_size) {
+        if (r->is_young()) {
+          young_region_ages.add(r->age(), r->get_live_data_words());
+        } else {
+          assert(r->is_old(), "Sanity");
+          old_region_ages.add(r->age(), r->get_live_data_words());
+        }
       }
     }
+    uint tenuring_threshold = heap->age_census()->tenuring_threshold();
+    st->print("Young regions: ");
+    young_region_ages.print_on(st, tenuring_threshold);
+    st->cr();
+    st->print("Old regions: ");
+    old_region_ages.print_on(st, tenuring_threshold);
```

Comment:
`tenuring_threshold` in this context doesn't make any sense for old regions.

Reply:
Changed to just use the max tenuring threshold here, although that doesn't really make sense here either. We can rethink this in the future, or elide that part. Leaving as is (using max tenuring threshold) for the moment.